### DISCUSSION TITLE
common-tool/scanner

### DISCUSSION THREAD

**Discussion by giscus** [itwanger/tobebetterjavaer-giscus#218] [Announcements]
# common-tool/scanner

本文深入剖析了Java中的Scanner类，详细介绍了其用法、功能以及如何在实际应用中扫描控制台输入。通过具体代码示例，让您更好地理解Scanner的工作原理，使得控制台输入处理变得简单高效。掌握Scanner类，让Java编程更加得心应手。

https://tobebetterjavaer.com/common-tool/scanner.html

**Comment by KayunZhang** [comment]
二哥
System.out.print("请输入多个单词，以空格分隔：");
while (scanner.hasNext()) { // 判断输入流中是否还有下一个标记
    String word = scanner.next(); // 读取输入流中的下一个单词
    System.out.println("您输入的单词是：" + word);
}
scanner.nextLine(); // 读取输入流中的换行符
这段代码会死循环，应该使用hasNext()的重载方法hasNext(String patten)加个标识符，比如#，遇到这个字符就结束循环：
System.out.print("请输入多个单词，以空格分隔,以'#'结束：");
        // 判断输入流中是否还有下一个标记
        while (!scanner.hasNext("#")) {
            // 读取输入流中的下一个单词
            String word = scanner.next();
            System.out.println("您输入的单词是：" + word);
        }
        // 读取输入流中的换行符
        scanner.nextLine();
这是我的代码  不过还有点问题  如果单独输入#号会跳过输入实数的部分，直接进入输入字符串，二哥帮忙解决一下

**Comment by RZG990813** [comment]
Java Scanner：金控控制台输入，还能扫描文件
文章标题 有错别字, 二哥,
